src/utils/classifier_report.py

```diff
@@ -231,6 +258,122 @@ def create_html_page(self, data, csv_path, stats=False):
                                 toggle_button_txt();
                                 };
                            ''')
+                        # updates review export button
+                        a('''
+                          let updateReviewButton = () => {
+                            let reviewItems = localStorage.getItem("reviewItems");
+                            let len = (reviewItems == null) ? 0 : JSON.parse(reviewItems).length;
+                            $("#reviewExportButton").text(`Export ${len} reviewed items`)
+                            if (len == 0) {
+                                    $("#reviewExportButton").prop("disabled",true);
+                                    $("#buttonsBottom").css("visibility", "hidden");
+
+                            } else {
+                                $("#reviewExportButton").prop("disabled",false);
+                                $("#buttonsBottom").css("visibility", "visible");
+
+                            }
+                          }
+                           ''')
+                        # loads review choices from local storage
+                        a('''
+                          let loadReviewChoices = () => {
+                          if (!localStorage.getItem('reviewItems')) {
+                            localStorage.setItem("reviewItems", "[]");  
+                          } else {
+                            let reviewedItems = JSON.parse(localStorage.getItem("reviewItems"))
+                            $('.reviewButton').removeClass('active');
+                            reviewedItems.forEach(e => {
+                                e = JSON.parse(e);
+                               if ($(`#${e['id']}`).hasClass('active') === false) {
+                                  $(`#${e['id']}`).toggleClass('active');
+                                }
+                            })
+                          }
+                        }
+                           ''')
+                        # initialize localStorage object
+                        a('''
+                        if (localStorage.length == 0) {
+                            localStorage.setItem("reviewItems", "[]");  
+                            }
+                        updateReviewButton();
+                           ''')
+                        # runs update functions
+                        a('''
+                          loadReviewChoices();
+                          updateReviewButton();
+                          ''')
+                        # functions to add or remove reviewed items to local storage
+                        a('''
+                            let removeReviewItem = (id) => {
+                                let reviewItems = JSON.parse(localStorage.getItem("reviewItems"));
+                                reviewItems = $.grep(reviewItems, (x) => JSON.parse(x).id != id);

```

Comment:
Just wondering, was there a reason for building a new Object with the key removed via grep/filtering here, versus using the `delete` function? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete

Reply:
Since local storage only saves values as strings, so I'd have to convert each item in the reviewItems array to an object then filter by id to find the correct object to delete. Using delete would require more lines of code, I believe.